dev/tests/integration/testsuite/Magento/Catalog/_files/my_product_with_options.php

```diff
@@ -0,0 +1,218 @@
+            ],
+            [
+                'title' => 'radio option 2',
+                'price' => 20,
+                'price_type' => 'fixed',
+                'sku' => 'radio option 2 sku',
+                'sort_order' => 2,
+            ],
+        ],
+    ],
+    [
+        'title' => 'checkbox option',
+        'type' => 'checkbox',
+        'is_require' => true,
+        'sort_order' => 6,
+        'values' => [
+            [
+                'title' => 'checkbox option 1',
+                'price' => 10,
+                'price_type' => 'fixed',
+                'sku' => 'checkbox option 1 sku',
+                'sort_order' => 1,
+            ],
+            [
+                'title' => 'checkbox option 2',
+                'price' => 20,
+                'price_type' => 'fixed',
+                'sku' => 'checkbox option 2 sku',
+                'sort_order' => 2,
+            ],
+        ],
+    ],
+    [
+        'title' => 'multiple option',
+        'type' => 'multiple',
+        'is_require' => true,
+        'sort_order' => 7,
+        'values' => [
+            [
+                'title' => 'multiple option 1',
+                'price' => 10,
+                'price_type' => 'fixed',
+                'sku' => 'multiple option 1 sku',
+                'sort_order' => 1,
+            ],
+            [
+                'title' => 'multiple option 2',
+                'price' => 20,
+                'price_type' => 'fixed',
+                'sku' => 'multiple option 2 sku',
+                'sort_order' => 2,
+            ],
+        ],
+    ],
+    [
+        'title' => 'date option',
+        'type' => 'date',
+        'price' => 80.0,
+        'price_type' => 'fixed',
+        'sku' => 'date option sku',
+        'is_require' => true,
+        'sort_order' => 8
+    ],
+    [
+        'title' => 'date_time option',
+        'type' => 'date_time',
+        'price' => 90.0,
+        'price_type' => 'fixed',
+        'is_require' => true,
+        'sort_order' => 9,
+        'sku' => 'date_time option sku'
+    ],
+    [
+        'title' => 'time option',
+        'type' => 'time',
+        'price' => 100.0,
+        'price_type' => 'fixed',
+        'is_require' => true,
+        'sku' => 'time option sku',
+        'sort_order' => 10
+    ]
+];
+
+$customOptions = [];
+
+/** @var ProductCustomOptionInterfaceFactory $customOptionFactory */
+$customOptionFactory = $objectManager->create(ProductCustomOptionInterfaceFactory::class);
+
+foreach ($options as $option) {
+    /** @var \Magento\Catalog\Api\Data\ProductCustomOptionInterface $customOption */
+    $customOption = $customOptionFactory->create(['data' => $option]);
+    $customOption->setProductSku($product->getSku());
+    $customOptions[] = $customOption;
+}
+
+$product->setOptions($customOptions);
+
+/** @var ProductRepositoryInterface $productRepository */
+$productRepository = $objectManager->get(ProductRepositoryInterface::class);
```

Comment:
Please move object creating to top of file (after object manager created) in whole file.